spylib/request.py

```diff
@@ -101,8 +99,33 @@ def _set_refresh_token(self, refresh_token):
         self.refresh_token = refresh_token
         self.notify_observers()
 
+    def can_authenticate(self):
+        """
+        Checks if the `ServiceRequestFactory` instance can authenticate.
+
+        Returns:
+            bool: Whether or not the instance is capable of authenticating.
+        """
+        if self.refresh_token is not None:
+            return True
+        
+        if (self.uuid is not None) and (self.api_key is not None):
+            return True
+        
+        return False
```

Comment:
Would we want to return true in the case that a user has an `access_token`, but, no `refresh_token`, `uuid`, or `api_key`?

Reply:
actually nm, that won't matter here I think 👍

Reply:
I think that would be good if the method was `is_authenticated`, but this specifically is checking if the request factory _can_ authenticate.

Reply:
What do you think?

Reply:
I think this works. 🎏